8.x/core-concepts/annotations.md

```diff
@@ -2,4 +2,423 @@
+```
+
+```php{1-2}
+$reflective_method->getComment();
+//returns the phpdoc on top of method
+```
+
+```php{1-2}
+$reflective_class->getAnnotations();
+//returns a list of annotations
+```
+
+```php{1-2}
+$reflective_class->hasAnnotation("specific_annotation");
+//checks if an specific annotation exists
+```
+
+```php{1-2}
+$reflective_class->getAnnotation(("specific_annotation"));
+//returns the specific annotation with passed title
+```
+
+## Examples of the use of annotation-parser package
+
+Make a new command, named `TestAnnotations`:
+
+```bash{1}
+php artisan make:command TestAnnotations
+```
+
+Now you have the codes below in the path `/larammerce/app/Console/Commands/TestAnnotations.php`:
+
+```php{1-42}
+<?php
+
+namespace App\Console\Commands;
+use Illuminate\Console\Command;
+
+class TestAnnotations extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'command:name';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Command description';
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        parent::__construct();
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return int
+     */
+    public function handle()
+    {
+        return 0;
+    }
+}
+```
+
+Change this line into the `TestAnnotations class`:
+
+```php{8}
+class TestAnnotations extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'larammerce:annotation';
+    ...
+}
+```
+
+Insert this code into the `public function handle`:
+
+```php{3}
+public function handle()
+{
+    dd("salam");
+}
+```
+
+Run the following command into the terminal:
+
+```bash{1}
+php artisan larammerce:annotation
+```
+
+**OUTPUT**
+
+![handle-salam](/02.png)
+
+In the path `/larammerce/app/Console/Commands/TestAnnotations.php`, construct a `ReflectiveClass` from the `TestAnnotations class` and get the DocBlock above this class:
+
+
+```php{5,12,13}
+<?php
+
+namespace App\Console\Commands;
+use Illuminate\Console\Command;
+use App\Utils\Reflection\ReflectiveClass;
+
+class TestAnnotations extends Command
+{
+    ...
+    public function handle()
+    {
+        $testAnnotationsRef = new ReflectiveClass(TestAnnotations::class);
+        dd($testAnnotationsRef->getComment());
+    }
+}
+```
+
+**OUTPUT**
+
+![no-docblock](/03.png)
+
+This output indicates that there is no DocBlock above the TestAnnotations class:
+
+```php{1}
+//No DocBlock here.
+class TestAnnotations extends Command
+{
+    ...
+}
+```
+
+Now put this DocBlock above the TestAnnotations class:
+
+```php{1-3}
+/**
+ * @salam(ali="2")
+ */
+class TestAnnotations extends Command
+{
+    ...
+}
+```
+
+**OUTPUT**
+
+s![mere-comment](/04.png)
```

Comment:
"s" should be a writing mistake here. please consider solving it.

Reply:
Corrected.